test/test_source_comment_associator.rb

```diff
@@ -91,7 +187,7 @@ def foo
 end
     END
 
-    assert_equal 0, associations.size
+    assert_equal 1, associations.size
```

Comment:
Maybe add a test which verifies that the stray comment is associated to the right node.